### ISSUE TITLE
`omc logs` always tries to open the insecure previous logs

### ISSUE THREAD

**Issue by palonsoro** [gmeghnag/omc#178]
If I run `omc logs -p`, it tries to open both `previous` and `previous.insecure` logs regardless of `--insecure` flag.

It should behave like `oc logs`, where the insecure logs are only displayed if `--insecure` is provided (if it is desired to warn the user that they can check the insecure logs, a warning in stderr might work well)

**Comment by bverschueren** [comment]
AFAICS `omc`'s `--insecure` flag is just short for `oc`/`kubectl`'s [InsecureSkipTLSVerifyBackend](https://github.com/kubernetes/kubernetes/blob/7436ca32bc766ff202109a7541d2e7bb41ee7d13/pkg/apis/core/types.go#L5489-L5496) flag. So in theory I guess this also applies on current or even rotated logs?

**Comment by palonsoro** [comment]
Yep.

More concretely: must-gather stores the `current.log` and `previous.log` files with the logs of the current (`oc logs` without `-p`) and previous  (`oc logs -p`) container logs. If either of them fails, it retries with `--insecure-skip-tls-verify-backend` to use the `InsecureSkipTLSVerifyBackend` you mentioned and stores thems in `current.insecure.log` and/or `previous.insecure.log` respectively. Then, the `--insecure` option of `omc logs` allows checking these "insecure" log files from the must-gather (check [this issue](https://github.com/gmeghnag/omc/issues/83) for the RFE and the commit that led to this).

Problem is that now, for some reason, omc seems to try to access both logs regardless of `--insecure` parameter being present or absent.

**Comment by cchen666** [comment]
It appears to me that usually the `oc adm inspect ns/XXX` will only collect previous.insecure.log (not current.insecure.log) because Pod's previous.log is often "Not Found". 

```
I0828 02:49:26.303994 1777166 request.go:1171] Response Body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"previous terminated container \"kube-apiserver-operator\" in pod \"kube-apiserver-operator-6d5bc8484-4qt4v\" not found","reason":"BadRequest","code":400}
I0828 02:49:26.304407 1777166 round_trippers.go:466] curl -v -XGET  -H "Accept: application/json, */*" -H "User-Agent: oc/4.13.0 (linux/amd64) kubernetes/17b7acc" 'https://api.sno-pek.cchen.work:6443/api/v1/namespaces/openshift-kube-apiserver-operator/pods/kube-apiserver-operator-6d5bc8484-4qt4v/log?container=kube-apiserver-operator&insecureSkipTLSVerifyBackend=true&previous=true&timestamps=true'
```

https://github.com/openshift/oc/blob/c72b183b6b632269d0210e321931b8fd02fab803/pkg/cli/admin/inspect/pod.go#L259-L266

But I think there is some case as Pablo mentions, that both previous.log and previous.insecure.log exist at the same time with valid logs. Anyway if we don't see current.log and current.insecure.log co-exist at the same time in any inspect or must-gather, perhaps we simply handle the --insecure option for previous.[.insecure].log only, not for current.log maybe?

```
func (l *LogReader) FromPrevious() {
	l.files = &[]string{previousLogFile, previousInsecureLogFile}
}
```

**Comment by bverschueren** [comment]
Yes, my bad. When [refactoring](https://github.com/gmeghnag/omc/commit/8562f9dc1c48e271468bf8ac8c3f95909d6078c3) the pod logs logic, the insecure logs became default part of `--previous / -p` output (if existed). This should be fixed in #179 for current/previous (and even rotated if that follows the convention).

**Comment by palonsoro** [comment]
@cchen666 insecure logs are gathered when gathering normal ones fails. So if there is a current.insecure, the current gathering failed and there is a previous.insecure, previous gathering failed.

It is more usual to see previous.insecure because it tries it the previous gathering failed due to no previous logs. However, there are situations where both current.insecure and previous.insecure are gathered and valid, typically when there is some certificate mismatch between kube-apiserver and kubelet that makes TLS validation fail, so we need to skip the validation via insecure. 

So we need to handle the option for both current and insecure. Mainly because that mimics oc behavior, but also because we need to distinguish if the log was gathered insecure or not
